pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java

```diff
@@ -53,21 +54,26 @@ public PinotTableRestletResource() throws IOException {
   @Override
   @Post("json")
   public Representation post(Representation entity) {
-    AbstractTableConfig config = null;
     try {
       String jsonRequest = entity.getText();
-      config = AbstractTableConfig.init(jsonRequest);
+      AbstractTableConfig config = AbstractTableConfig.init(jsonRequest);
       try {
         addTable(config);
       } catch (Exception e) {
-        LOGGER.error("Caught exception while adding table", e);
-        ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(ControllerMeter.CONTROLLER_TABLE_ADD_ERROR, 1L);
-        setStatus(Status.SERVER_ERROR_INTERNAL);
+        if (e instanceof PinotHelixResourceManager.InvalidTableConfigException) {
+          LOGGER.info("Invalid table config, {}", e.getMessage());
+          setStatus(Status.CLIENT_ERROR_BAD_REQUEST);
+        } else {
+          LOGGER.error("Caught exception while adding table", e);
+          setStatus(Status.SERVER_ERROR_INTERNAL);
+        }
+        ControllerRestApplication.getControllerMetrics()
+            .addMeteredGlobalValue(ControllerMeter.CONTROLLER_TABLE_ADD_ERROR, 1L);
         return new StringRepresentation("Failed: " + e.getMessage());
       }
       return new StringRepresentation("Success");
     } catch (Exception e) {
-      LOGGER.error("error reading/serializing requestJSON", e);
+      LOGGER.info("Caught exception while deserializing table config", e);
```

Comment:
This one we are unable to log table name because it's de-serialization exception.